libparsec/crates/platform_mountpoint/src/windows/filesystem.rs

```diff
@@ -49,20 +49,51 @@ fn is_entry_info_path(path: &FsPath) -> bool {
         .unwrap_or_default()
 }
 
+/// In Windows logic opening a file/folder provides an exclusive handle
+/// on it so that no concurrent modification can occur.
+///
+/// `OpenedObj` represents such opened resource.
+///
+/// Note there is a trick here, as in Parsec not all modifications go through
+/// the filesystem (i.e. GUI and remote changes), so a concurrent modification
+/// may move the resource (or event modify its type !).
 #[derive(Debug)]
 pub(crate) enum OpenedObj {
     Folder {
         parsec_file_name: FsPath,
+        id: VlobID,
         folder_reader: Option<FolderReader>,
     },
     File {
+        // Keeping file name here is useful for debug logs
+        #[allow(dead_code)]
         parsec_file_name: FsPath,
+        #[allow(dead_code)]
+        id: VlobID,
         fd: FileDescriptor,
     },
-    // EntryInfo {
-    //     parsec_file_name: FsPath,
-    //     info: FileInfo,
-    // },
+}
+
+fn parsec_file_stat_to_winfsp_file_info(stat: &FileStat) -> FileInfo {
+    let created = filetime_from_utc((stat.created).into());
+    let updated = filetime_from_utc((stat.updated).into());
+    *FileInfo::default()
+        // FILE_ATTRIBUTE_ARCHIVE is a good default attribute
+        // This way, we don't need to deal with the weird semantics of
+        // FILE_ATTRIBUTE_NORMAL which means "no other attributes is set"
+        // Also, this is what the winfsp memfs does.
+        .set_file_attributes(FileAttributes::ARCHIVE | FileAttributes::NOT_CONTENT_INDEXED)
+        .set_creation_time(created)
+        .set_last_access_time(updated)
+        .set_last_write_time(updated)
+        .set_change_time(updated)
+        // TODO: We truncate the EntryID to 64bits as index number, given EntryID is a UUIDv4 this
+        // increases the risk of collision... We should investigate to see if this is really needed.
+        .set_index_number(stat.id.as_u128() as u64)
```

Comment:
Should you be more explicit on the truncate between little/big-endian ?

Reply:
I have added a comment :+1: